crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -56,10 +57,20 @@ pub async fn run() -> anyhow::Result<()> {
 }
 
 impl KvRedisProvider {
+    pub fn name() -> &'static str {
+        "keyvalue-redis-provider"
+    }
+
     pub async fn run() -> anyhow::Result<()> {
+        initialize_observability!(
+            KvRedisProvider::name(),
+            std::env::var_os("PROVIDER_KV_REDIS_FLAMEGRAPH_PATH")
```

Comment:
Do we actually pass this environment variable down to the provider?

Reply:
Yeah this is for the case someone is running the provider locally (and trying to debug it), less so for when it's running in the host just normally. 

That said, we should probably have a way to pass more than a few env vars down to the child processes for providers!

Reply:
Hmnnn maybe it's better to get that value from `HostData` *and* ENV (and use whichever is specified). Having an example of how to do that or a utility function might be quite nice.